mirgecom/multiphysics/thermally_coupled_fluid_wall.py

```diff
@@ -150,13 +174,78 @@ def get_external_grad_av(self, discr, dd_bdry, grad_av_minus, **kwargs):
         # Grab a unit normal to the boundary
         nhat = thaw(discr.normal(dd_bdry), actx)
 
+        # REVIEW QUESTION
         # Apply a Neumann condition on the energy gradient
-        ext_grad_energy = (
-            grad_av_minus.energy - 2 * np.dot(grad_av_minus.energy, nhat) * nhat)
+        # This also looks suspiscous: I blv the "2" should not be there
+        # for this term.
+        ext_grad_energy = \
+            grad_av_minus.energy - 2 * np.dot(grad_av_minus.energy, nhat) * nhat
+
+        # REVIEW QUESTION
+        # uh oh - we don't have the necessary data to compute grad_y from grad_av
+        # from mirgecom.fluid import species_mass_fraction_gradient
+        # grad_y_minus = species_mass_fraction_gradient(state_minus.cv,
+        #                                               grad_cv_minus)
+        # grad_y_plus = grad_y_minus - np.outer(grad_y_minus@normal, normal)
+        # grad_species_mass_plus = 0.*grad_y_plus
+        # This re-stuffs grad_y+ back into grad_cv+, skipit; we did not split AVs
+        # for i in range(state_minus.nspecies):
+        #    grad_species_mass_plus[i] = (state_minus.mass_density*grad_y_plus[i]
+        #        + state_minus.species_mass_fractions[i]*grad_cv_minus.mass)
+        ext_grad_species_mass = \
+            grad_av_minus.species_mass - np.outer(grad_av_minus@nhat, nhat)
```

Comment:
In order to enforce `grad(Y) .dot. nhat = 0` on the boundary, we must explicitly _set_ this to be true in grad(Q)_+ , because we compute the viscous flux at the boundary face as just `F_v = viscous_flux(Q_+, grad(Q)_+)`.  But the `viscous_flux` function computes `grad(Y)` using the product rule down inside that function.    So in the boundary treatment, we compute `grad(Y)`, set it's wall-normal component to 0, and then stuff it back into `grad(Q)_+` so that when `F_v` is computed down inside `viscous_flux` - it gets 0 for the wall-normal component of `grad(Y)`.

See here to see how that is getting done for our viscous flux in the IsothermalWall:
https://github.com/illinois-ceesd/mirgecom/blob/1eb78355a414bb79340117c7c7b84a8e423cc065/mirgecom/boundary.py#L860

In _this_function, (i.e. the AV function in discussion) we don't actually have Q, and grad(Q), we have the AV solution and gradient, so we can't do the same sort of thing.    Also, maybe we shouldn't spend a gob of time trying to understand what to do for AV since we plan to abandon it pdq.   I think if AV is just using like an average or some function of both flux(+) and flux(-), then we should fix it up s.t. the average is what we intend to enforce at the boundary.

Reply:
Ah. Yeah, I'm on board with just finding something that works for now and rolling with it. The Neumann condition I'm specifying for the energy gradient here isn't really correct either IIRC.